diff -urNp -x '*.orig' monit-5.25.2.org/monitrc monit-5.25.2/monitrc
--- monit-5.25.2.org/monitrc	2018-05-29 21:03:45.000000000 +0200
+++ monit-5.25.2/monitrc	2021-10-05 22:43:27.407462292 +0200
@@ -76,7 +76,8 @@ set log syslog
 ## specified using a comma separator. If the first mail server fails, Monit
 # will use the second mail server in the list and so on. By default Monit uses
 # port 25 - it is possible to override this with the PORT option.
-#
+
+set mailserver localhost
 # set mailserver mail.bar.baz,               # primary mailserver
 #                backup.bar.baz port 10025,  # backup mailserver on port 10025
 #                localhost                   # fallback relay
@@ -137,9 +138,9 @@ set log syslog
 ## You can set alert recipients whom will receive alerts if/when a
 ## service defined in this file has errors. Alerts may be restricted on
 ## events by using a filter as in the second example below.
-#
-# set alert sysadm@foo.bar                       # receive all alerts
-#
+
+set alert root@localhost                        # receive all alerts
+
 ## Do not alert when Monit starts, stops or performs a user initiated action.
 ## This filter is recommended to avoid getting alerts for trivial cases.
 #
@@ -300,6 +301,5 @@ set httpd port 2812 and
 ##
 ## It is possible to include additional configuration parts from other files or
 ## directories.
-#
-#  include /etc/monit.d/*
-#
+
+include /etc/monit/*.monitrc
