diff -ur monit-4.8/util.c monit-4.8-n/util.c
--- monit-4.8/util.c	2006-04-27 14:16:03.000000000 -0600
+++ monit-4.8-n/util.c	2006-10-04 10:35:39.000000000 -0600
@@ -1738,16 +1738,11 @@
  * @return the hostBuffer
  */
 char *Util_getHTTPHostHeader(Socket_T s, char *hostBuf, int len) {
-  if(! strcmp(LOCALHOST, socket_get_remote_host(s)) ||
-     inet_aton(socket_get_remote_host(s), NULL)) {
-    *hostBuf= 0;
-  } else {
-    if(socket_get_remote_port(s)==80)
-       snprintf(hostBuf, len, "%s", socket_get_remote_host(s));
-     else
-       snprintf(hostBuf, len, "%s:%d", socket_get_remote_host(s),
-                socket_get_remote_port(s));
-  }
+  if(socket_get_remote_port(s)==80)
+     snprintf(hostBuf, len, "%s", socket_get_remote_host(s));
+   else
+     snprintf(hostBuf, len, "%s:%d", socket_get_remote_host(s),
+              socket_get_remote_port(s));
   return hostBuf;
 }
 
